packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmObjectHelper.kt

```diff
@@ -643,19 +683,26 @@ internal object RealmObjectHelper {
         }
     }
 
-    // TODO remove UnusedPrivateMember when parameter is used
-    @Suppress("UnusedPrivateMember")
     internal inline fun <reified T : Any> setDictionary(
         obj: RealmObjectReference<out BaseRealmObject>,
         col: String,
         dictionary: RealmDictionary<T>,
         updatePolicy: UpdatePolicy = UpdatePolicy.ALL,
         cache: UnmanagedToManagedObjectCache = mutableMapOf()
     ) {
-        TODO()
+        val existingDictionary = getDictionary<T>(obj, col)
+        if (dictionary !is ManagedRealmDictionary<T> || !RealmInterop.realm_equals(
```

Comment:
Maybe I'm missing it, but we only enter the if statement if it evaluates to true, so if dictionary is ManagedRealmDictionary it will skip it, doing nothing?

Reply:
The second part of the `if` statement is only checked if the first part is not satisfied, i.e. `dictionary !is ManagedRealmDictionary<T>` which is not true in the case you mention because the right-hand side of the assignment is a managed dictionary. Then, the second part of the `if` kicks in to compare whether the pointers of both managed dictionaries are the same, which are not, so the receiver dictionary is updated correctly.